readthedocs/v3/serializers.py

```diff
@@ -0,0 +1,429 @@
+        return programming_language
+
+    def get_name(self, programming_language):
+        for code, name in PROGRAMMING_LANGUAGES:
+            if code == programming_language:
+                return name
+        return 'Unknown'
+
+
+class ProjectURLsSerializer(serializers.Serializer):
+    documentation = serializers.CharField(source='get_docs_url')
+    project = serializers.SerializerMethodField()
+
+    def get_project(self, obj):
+        # Overridden only to return ``None`` when the description is ``''``
+        return obj.project_url or None
+
+
+class RepositorySerializer(serializers.Serializer):
+
+    url = serializers.CharField(source='repo')
+    type = serializers.CharField(source='repo_type')
+
+
+class ProjectLinksSerializer(BaseLinksSerializer):
+
+    _self = serializers.SerializerMethodField()
+
+    # TODO: add these once the endpoints get implemented
+    # users = serializers.SerializerMethodField()
+    versions = serializers.SerializerMethodField()
+    builds = serializers.SerializerMethodField()
+    subprojects = serializers.SerializerMethodField()
+    superprojects = serializers.SerializerMethodField()
+    translations = serializers.SerializerMethodField()
+
+    def get__self(self, obj):
+        path = reverse('projects-detail', kwargs={'project_slug': obj.slug})
+        return self._absolute_url(path)
+
+    def get_versions(self, obj):
+        path = reverse(
+            'projects-versions-list',
+            kwargs={
+                'parent_lookup_project__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_builds(self, obj):
+        path = reverse(
+            'projects-builds-list',
+            kwargs={
+                'parent_lookup_project__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_subprojects(self, obj):
+        path = reverse(
+            'projects-subprojects',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_superprojects(self, obj):
+        path = reverse(
+            'projects-superprojects',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_translations(self, obj):
+        path = reverse(
+            'projects-translations',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class ProjectSerializer(FlexFieldsModelSerializer):
+
+    language = LanguageSerializer()
+    programming_language = ProgrammingLanguageSerializer()
+    repository = RepositorySerializer(source='*')
+    privacy_level = PrivacyLevelSerializer(source='*')
+    urls = ProjectURLsSerializer(source='*')
+    subproject_of = serializers.SerializerMethodField()
+    translation_of = serializers.SerializerMethodField()
```

Comment:
I think we call these `main_project` and `main_language` internally

Reply:
Yes. I think that `subproject_of` and `translation_of` are better names for the users of the API, though.